File: 1545-find-kth-bit-in-nth-binary-string/1545-find-kth-bit-in-nth-binary-string.cpp
class Solution {
public:
    char findKthBit(int n, int k) {
        string s = "0";

        for(int i = 1; i < n; i++)
        {
            string add = invert(s);
            reverse(add.begin(), add.end());
            s = s + '1' + add;
        }
        return s[k - 1];
    }
private:
    string invert(string& s)
    {
        string res = "";
        for(int i = 0; i < s.length(); i++)
        {
            if(s[i] == '0')
            {
                res += '1';
            }
            else
            {
                res += '0';
            }
        }
        return res;
    }
};

File: 1545-find-kth-bit-in-nth-binary-string/1545-find-kth-bit-in-nth-binary-string.py
class Solution:
    def findKthBit(self, n: int, k: int) -> str:
        if n == 1:
            return '0'
        length = (1 << n) - 1
        if k - 1 == length // 2:
            return '1'
        elif k - 1 < length // 2:
            return self.findKthBit(n - 1, k)
        return '0' if self.findKthBit(n - 1, length - k + 1) == '1' else '1'